homeassistant/components/brother/config_flow.py

```diff
@@ -64,6 +69,61 @@ class BrotherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        try:
+            if not user_input.get("name").startswith("Brother"):
+                return self.async_abort(reason="not_brother_printer")
+        except AttributeError:
```

Comment:
It's weird that we except attribute error. Just add another check before the first check and combine with `or`.
```py
if not user_input.get("name") or not user_input["name"].startswith("Brother"):
    ...
```

Reply:
Done.